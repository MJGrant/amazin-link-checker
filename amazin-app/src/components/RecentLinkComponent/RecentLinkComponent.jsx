import React, { Component } from 'react'
import { Button } from 'reactstrap';

export default class RecentLinkComponent extends Component {

    static defaultProps = {
        recentLink: ''
    }

    render() {
        const { recentLink, index, handleClick } = this.props;

        return (
            <tr>
                <td className="tableUsersLinkText">
                    <a href={recentLink} id={"recent-url-" + index}>{recentLink}</a>
                </td>
                <td width='80'>
                    <Button
                        color="warning"
                        size="sm"
                        id={"button-retest-" + index}
                        onClick={handleClick}>
                        Re-test
                              </Button>
                </td>
            </tr>
        )
    }
}
