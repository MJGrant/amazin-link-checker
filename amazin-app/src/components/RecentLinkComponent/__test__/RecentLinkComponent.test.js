import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

import RecentLinkComponent from '../RecentLinkComponent'

describe('RecentLinkComponent', () => {
    const mockFunc = jest.fn();
    const mockLink = 'http://recent-link-component.com';

    test('renders itself with defaults', () => {
        const component = renderer.create(<RecentLinkComponent />)
        expect(component).toMatchSnapshot()
    })

    test('renders itself text prop', () => {
        const component = renderer.create(<RecentLinkComponent />)
        expect(component).toMatchSnapshot()
    })

    test('link text matches what was given', () => {
        const component = shallow(<RecentLinkComponent
            recentLink={mockLink}
            index={0}
            handleClick={mockFunc} />);
        expect(component.find('#recent-url-0').text()).toEqual(mockLink);
    });

    test('should call mock function when button is clicked', () => {
        const component = shallow(<RecentLinkComponent
            recentLink={mockLink}
            index={0}
            handleClick={mockFunc} />);
        const submitButton = component.find('#button-retest-0')
        submitButton.simulate('click');
        expect(mockFunc).toHaveBeenCalled();
    })
})