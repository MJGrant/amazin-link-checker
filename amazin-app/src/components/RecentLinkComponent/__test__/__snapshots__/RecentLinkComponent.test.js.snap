// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecentLinkComponent renders itself text prop 1`] = `
<tr>
  <td
    className="tableUsersLinkText"
  >
    <a
      href=""
      id="recent-url-undefined"
    >
      
    </a>
  </td>
  <td
    width="80"
  >
    <button
      aria-label={null}
      className="btn btn-warning btn-sm"
      id="button-retest-undefined"
      onClick={[Function]}
    >
      Re-test
    </button>
  </td>
</tr>
`;

exports[`RecentLinkComponent renders itself with defaults 1`] = `
<tr>
  <td
    className="tableUsersLinkText"
  >
    <a
      href=""
      id="recent-url-undefined"
    >
      
    </a>
  </td>
  <td
    width="80"
  >
    <button
      aria-label={null}
      className="btn btn-warning btn-sm"
      id="button-retest-undefined"
      onClick={[Function]}
    >
      Re-test
    </button>
  </td>
</tr>
`;
